test/sagemaker_tests/pytorch/training/integration/sagemaker/__init__.py

```diff
@@ -157,15 +157,19 @@ def _test_mnist_distributed(
     instance_groups=None,
     use_inductor=False,
 ):
-    dist_method = "pytorchddp" if dist_backend.lower() == "nccl" else "torch_distributed"
+    if dist_backend.lower() == "nccl":
```

Comment:
can you explain why for `nccl` we are using `smdistributed` launcher but otherwise we are using `torch_distributed`?

Reply:
I'm not entirely sure why it was originally using `pytorchddp` for nccl backend and `torch_distributed` when not. It came from this [PR](https://github.com/aws/deep-learning-containers/commit/53f99fafb1c247634df1eb80ca25587bf6b3d100). For the purpose of this PR, we are renaming all `pytorchddp` launcher to `smdistributed` launcher and only that to avoid breaking any changes.

Personally, I think for this specific test, we can just run `torch_distributed` only since we have dedicated tests for `smdistributed` and `pytorchddp` launcher already.